### ISSUE TITLE
rm -rf / actually deletes everything

### ISSUE THREAD

**Issue by Billy02357** [SerenityOS/serenity#5253]
executing ```rm -rf /``` as root deletes everything, maybe this should have some kind of protection so users don't delete their entire system by mistake

**Comment by danboid** [comment]
That's not a bug, it's a feature!

**Comment by tomuta** [comment]
> That's not a bug, it's a feature!

That's right, the `-f` means force...

Now it probably shouldn't delete mount points (e.g. `/proc`, `/dev`/ ...) that haven't been unmounted, not sure how other systems deal with that. I believe Linux would fail this despite `-f`. So, if we're nuking these as well without requiring an `unmount` first then I think this would be a bug.

**Comment by Billy02357** [comment]
This is exactly what I mean, yeah sure, I know the means forving it, but thats still something can happen acccidentally misstyping and delete everything. Should at least pop up a dialog

**Comment by tomuta** [comment]
Since `rm -rf` is commonly used in scripts that expect to work without user input, I don't think that's a good idea. Linux doesn't prompt for that very reason, either ([try in your browser](https://bellard.org/jslinux/vm.html?url=alpine-x86.cfg&mem=192)). It does fail to delete e.g. mount points that have not been unmounted, but continues on the deletion process.

**Comment by L0ric0** [comment]
but linux treats `/` special for rm:
```
--no-preserve-root
              do not treat '/' specially

       --preserve-root[=all]
              do not remove '/' (default); with 'all', reject any command line argument on a separate device from its parent
```
which was probably added because it happend far to often. Questin being if that sould also be added here.

**Comment by tomuta** [comment]
It doesn't treat it special unless you specify these options explicitly.

**Comment by L0ric0** [comment]
no `--preserve-root=/` is the default

**Comment by willmcpherson2** [comment]
For what it's worth, when I run `rm -rf /` I get:

```
[#0 rm(27:27)]: kmalloc(): Adding memory to heap at V0xc0801000, bytes: 1048576
[#0 rm(27:27)]: kmalloc(): Queue removing memory from heap at V0xc35f3000, bytes: 1048576
[#0 rm(27:27)]: kmalloc(): Queued memory region at V0xc35f3000, bytes: 1048576 will be freed now
```
`...`
```
[#0 rm(27:27)]: kmalloc(): Adding memory to heap at V0xc654e000, bytes: 1048576
[#0 rm(27:27)]: kmalloc(): Adding even more memory to heap at V0xc6b0c000, bytes: 41930752
[#0 rm(27:27)]: kmalloc(): Cannot expand heap: no backup memory
[rm(27:27)]: KERNEL PANIC! :^(
[rm(27:27)]: ASSERTION FAILED: index < this->m_size
[rm(27:27)]: .././AK/Span.h:206 in constexpr T& AK::Span<T>::at(size_t) [with T = const AK::TypeErasedParameter; size_t = long unsigned int]
[#0 rm(27:27)]: 0xc01189bf
[#0 rm(27:27)]: 0xc023f4e7
[#0 rm(27:27)]: 0xc023fc60
[#0 rm(27:27)]: 0xc0240de4
[#0 rm(27:27)]: 0xc011d929
[#0 rm(27:27)]: 0xc0251c56
[#0 rm(27:27)]: 0xc0150ee9
[#0 rm(27:27)]: 0xc014a2d2
[#0 rm(27:27)]: 0xc0148f91
[#0 rm(27:27)]: 0xc014d673
[#0 rm(27:27)]: 0xc017cbcd
[#0 rm(27:27)]: 0xc01f1ada
[#0 rm(27:27)]: 0xc01b894f
[#0 rm(27:27)]: 0xc01b7edd
```

Also an [interesting little tidbit](https://news.ycombinator.com/item?id=8896812) wrt POSIX compatibility.

**Comment by linusg** [comment]
Implemented in #11565 (whether people like it or not is a different question :^))
